### PR TITLE
[MM-43736] Auto follow threads on call join

### PR DESCRIPTION
**Pull request by Willyfrog**
#### Summary

When a user joins a call, automatically follow the thread to be notified of any coments to it.

CleanShot 2022-04-22 at 23 54 41

#### Ticket

[MM-43736](https://mattermost.atlassian.net/browse/MM-43736)

#### What's missing

Needs to have the same behavior on mobile. Probably next OSF

### CODE REVIEW

File: webapp/src/index.tsx [hunk 1]
```diff
@@ -545,7 +575,9 @@ export default class Plugin {
             const currentChannelId = getCurrentChannelId(store.getState());
             if (currChannelId !== currentChannelId) {
                 currChannelId = currentChannelId;
-                fetchChannelData(currChannelId);
+                fetchChannelData(currChannelId).then(() => {
+                    followThread(currChannelId, getCurrentTeamId(store.getState()));
```
Comment (streamer45) [inline webapp/src/index.tsx:579]: What's the reason for following a thread on channel switch?

File: webapp/src/components/channel_call_toast/component.tsx [hunk 2]
```diff
@@ -49,7 +50,7 @@ export default class ChannelCallToast extends React.PureComponent<Props, State>
         if (this.props.connectedID) {
             return;
         }
-        window.postMessage({type: 'connectCall', channelID: this.props.currChannelID}, window.origin);
+        window.postMessage({type: 'connectCall', channelID: this.props.currChannelID, teamID: this.props.currTeamId}, window.origin);
```
Comment (streamer45) [inline webapp/src/components/channel_call_toast/component.tsx:53]: Is this redundant? I see we are loading the team regardless of whether it's passed in the event handler.

**Review by streamer45 — COMMENTED** [review]
Thanks @Willyfrog, just a couple of questions. 

Also, could I ask you to create a ticket for this as I'd like to keep track of any "external" contribution? Thanks :)

File: webapp/src/index.tsx (hunk 1 above)
Comment (Willyfrog) [reply webapp/src/index.tsx:579]: that is a missunderdstanding on that part of the code, I thought it was switching from one call to other, not channels. I'll remove this.

File: webapp/src/components/channel_call_toast/component.tsx (hunk 2 above)
Comment (Willyfrog) [reply webapp/src/components/channel_call_toast/component.tsx:53]: Yes, it is, I'll remove this.

**Comment by Willyfrog** [comment]
> Also, could I ask you to create a ticket for this as I'd like to keep track of any "external" contribution? Thanks :)

https://mattermost.atlassian.net/browse/MM-43736

File: webapp/src/components/channel_call_toast/index.js [hunk 3]
```diff
@@ -28,6 +30,7 @@ const mapStateToProps = (state) => {
     }
     return {
         currChannelID: currentID,
+        currTeamId: teamID,
```
Comment (streamer45) [inline webapp/src/components/channel_call_toast/index.js:33]: Do we still need this?

File: webapp/src/index.tsx [hunk 4]
```diff
@@ -331,6 +342,7 @@ export default class Plugin {
 
                     if (!connectedChannelID(store.getState())) {
                         connectCall(channel.id);
+                        followThread(channel.id, channel.teamID);
```
Comment (streamer45) [inline webapp/src/index.tsx:345]: Should this be `channel.team_id` instead?

**Review by streamer45 — APPROVED** [review]
Looks good, thank you @Willyfrog :raised_hands:

File: webapp/src/index.tsx [hunk 5]
```diff
@@ -361,6 +373,15 @@ export default class Plugin {
             );
         };
 
+        const followThread = async (channelID: string, teamID: string) => {
+            const threadID = voiceChannelRootPost(store.getState(), channelID);
+            if (threadID) {
+                setThreadFollow(getCurrentUserId(store.getState()), teamID, threadID, true)(store.dispatch, store.getState);
```
Comment (cpoile) [inline webapp/src/index.tsx:379]: Nit: we could probably do this, right?
```suggestion
                store.dispatch(setThreadFollow(getCurrentUserId(store.getState()), teamID, threadID, true));
```

**Review by cpoile — APPROVED** [review]
Looks great @Willyfrog ! Just one minor suggestion for improvement.
Thanks for the contribution, this will make the UX nicer. :)

**Comment by streamer45** [comment]
Thanks @Willyfrog , this is good to merge as soon as CI passes.

**Pull request merged**